src/Basescape/BasescapeState.cpp

```diff
@@ -247,41 +251,22 @@ void BasescapeState::init()
  * Changes the base currently displayed on screen.
  * @param base Pointer to new base to display.
  */
-void BasescapeState::setBase(Base *base)
+void BasescapeState::setBase(entt::entity baseId)
 {
-	if (!getGame()->getSavedGame()->getBases().empty())
+	int index = 0;
+	auto baseView = getGame()->getSavedGame()->getRegistry().view<Base>().each();
+	for (auto&& [id, base] : baseView)
 	{
-		// Check if base still exists
-		bool exists = false;
-		for (size_t i = 0; i < getGame()->getSavedGame()->getBases().size(); ++i)
+		if (id == baseId)
 		{
-			if (getGame()->getSavedGame()->getBases().at(i) == base)
-			{
-				_base = base;
-				_mini->setSelectedBase(i);
-				getGame()->getSavedGame()->setSelectedBase(i);
-				exists = true;
-				break;
-			}
-		}
-		// If base was removed, select first one and reset index of visible bases
-		if (!exists)
-		{
-			_base = getGame()->getSavedGame()->getBases().front();
-			_mini->setSelectedBase(0);
-			getGame()->getSavedGame()->setSelectedBase(0);
-			_mini->setVisibleBasesIndex(0);
-			getGame()->getSavedGame()->setVisibleBasesIndex(0);
-
+			_base = &base;
+			_mini->setSelectedBaseIndex(index);
+			getGame()->getSavedGame()->setSelectedBaseIndex(index);
+			return;
 		}
+		++index;
 	}
-	else
-	{
-		// Use a blank base for special case when player has no bases
-		_base = new Base(getGame()->getMod());
-		_mini->setSelectedBase(0);
-		getGame()->getSavedGame()->setSelectedBase(0);
-	}
+	Log(LOG_ERROR) << "Base not found.";
```

Comment:
Need to set the base to the default base here to remain consistent with the original source.